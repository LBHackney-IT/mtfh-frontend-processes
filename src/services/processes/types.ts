export enum Trigger {
  CloseProcess = "CloseProcess",
  CancelProcess = "CancelProcess",
}
