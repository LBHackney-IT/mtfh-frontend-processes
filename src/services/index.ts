export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
