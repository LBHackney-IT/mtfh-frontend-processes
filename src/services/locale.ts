import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
