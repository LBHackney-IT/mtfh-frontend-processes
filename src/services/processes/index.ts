export * from "./sole-to-joint";
export * from "./change-of-name";
