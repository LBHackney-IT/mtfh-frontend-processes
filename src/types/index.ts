export type TargetType = "tenure" | "property" | "person";

export interface IStartProcess {
  thirdPartyCondition?: string;
  thirdPartyComponent?: React.FC;
  subHeading?: string;
  subComponent?: React.FC;
}

export interface IProcess {
  processName: string;
  title: string;
  targetType: TargetType;
  startProcess: IStartProcess;
  states: {
    [state: string]: {
      state: string;
      triggers: Record<string, string>;
    };
  };
}
