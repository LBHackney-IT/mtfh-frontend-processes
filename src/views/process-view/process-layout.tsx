import React, { useState } from "react";
import { Link as RouterLink, useParams } from "react-router-dom";

import {
  ChangeOfNameSideBar,
  ChangeOfNameView,
  SoleToJointSideBar,
  SoleToJointView,
} from "..";
import { CloseProcessDialog } from "../../components";
import { locale, processes } from "../../services";
import { CommentsView } from "./shared/comments-view/comments-view";

import { useProcess } from "@mtfh/common/lib/api/process/v1";
import {
  Center,
  ErrorSummary,
  Heading,
  Layout,
  Link,
  Spinner,
} from "@mtfh/common/lib/components";

const components = {
  [processes.soletojoint.processName]: SoleToJointView,
  [processes.changeofname.processName]: ChangeOfNameView,
};

const sidebars = {
  [processes.soletojoint.processName]: SoleToJointSideBar,
  [processes.changeofname.processName]: ChangeOfNameSideBar,
};

const isSameState = (firstState, secondState) => {
  return firstState.state === secondState.state;
};

export const ProcessLayout = (): JSX.Element => {
  const { processId, processName } =
    useParams<{ processId: string; processName: string }>();

  const [isCloseProcessDialogOpen, setCloseProcessDialogOpen] = useState<boolean>(false);
  const [closeProcessReason, setCloseProcessReason] = useState<string>();
  const [isCancel, setCancel] = useState<boolean>(false);
  const [submitted, setSubmitted] = useState<boolean>(false);
  const [closeCase, setCloseCase] = useState<boolean>(false);

  const {
    data: process,
    error,
    mutate,
  } = useProcess({
    id: processId,
    processName,
  });

  if (error) {
    return (
      <ErrorSummary
        id="sole-to-joint-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!process) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const Component = components[processName];
  const Sidebar = sidebars[processName];

  if (!Component) {
    return (
      <ErrorSummary
        id="process-layout-error"
        title={locale.errors.unableToFindState}
        description={locale.errors.unableToFindStateDescription}
      />
    );
  }

  const processConfig = processName && processes[processName];
  const { states } = processConfig;
  const { processClosed } = states;

  let closeProcessReasonFinal = closeProcessReason;
  if (!closeProcessReasonFinal && isSameState(process.currentState, processClosed)) {
    closeProcessReasonFinal = process.currentState.processData.formData.Reason;
  }

  return (
    <Layout
      data-testid="process-layout"
      sidePosition="right"
      backLink={
        <Link
          as={RouterLink}
          to={`/processes/${processName}/start/${process.targetType}/${process.targetId}`}
          variant="back-link"
        >
          {locale.backButton}
        </Link>
      }
      side={
        <Sidebar
          process={process}
          submitted={submitted}
          closeCase={closeCase}
          setCloseProcessDialogOpen={setCloseProcessDialogOpen}
          setCancel={setCancel}
        />
      }
    >
      <Heading variant="h1">{processConfig.title}</Heading>
      <Component
        process={process}
        mutate={mutate}
        optional={{
          submitted,
          setSubmitted,
          closeCase,
          setCloseCase,
          setCancel,
          setCloseProcessDialogOpen,
          closeProcessReasonFinal,
        }}
      />

      <CloseProcessDialog
        isCloseProcessDialogOpen={isCloseProcessDialogOpen}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCloseProcessReason={setCloseProcessReason}
        mutate={mutate}
        isCancel={isCancel}
      />

      <hr className="divider" />
      <CommentsView targetType="process" targetId={processId} mutate={mutate} />
    </Layout>
  );
};
