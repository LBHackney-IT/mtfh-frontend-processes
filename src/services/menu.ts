import * as processes from "./processes";

import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";

interface MenuProps {
  label: string;
  link: string;
  processes?: { label: string; link: string; targetType: string }[];
  getPplQuery?: (context: any) => string;
}

const getProcessLink = (processName: string) => {
  const process = processes[processName];
  return {
    targetType: process.targetType,
    label: process.title,
    link: `/processes/${process.processName}/start`,
  };
};

const getPplQueryForRequestALegalReferral = ({ tenure, asset }): string => {
  if (!tenure || !asset) {
    return "";
  }
  const {
    householdMembers,
    tenuredAsset: { propertyReference },
  } = tenure as Tenure;
  const {
    assetAddress: { addressLine1, postCode },
  } = asset as Asset;
  const tenantNames = householdMembers
    .filter((member) => member.isResponsible)
    .map((member) => member.fullName)
    .join(", ");
  const address = addressLine1;
  const query = `usp=pp_url&entry.1699199605=${tenantNames}&entry.2086190845=${address}&entry.1956036523=${postCode}&entry.1140527432=${propertyReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const getPplQueryTenancyChange = ({ tenure }): string => {
  if (!tenure) {
    return "";
  }
  const {
    householdMembers,
    paymentReference,
    tenuredAsset: { fullAddress, propertyReference },
  } = tenure as Tenure;

  const tenant = householdMembers.find((member) => member.isResponsible);
  const query = `usp=pp_url&entry.1942863541=${tenant?.fullName}&entry.1090481808=${fullAddress}&entry.321634337=${propertyReference}&entry.582205322=${paymentReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const menu: MenuProps[] = [
  {
    label: "Sole to Joint",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    processes: [
      getProcessLink(processes.soletojoint.processName),
      getProcessLink(processes.changeOfName.processName),
    ],
  },
  {
    label: "Other tenancy changes",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    getPplQuery: getPplQueryTenancyChange,
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
    getPplQuery: getPplQueryForRequestALegalReferral,
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export default menu;
