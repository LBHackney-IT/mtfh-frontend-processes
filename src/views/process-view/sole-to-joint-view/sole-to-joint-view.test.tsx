import React, { useState } from "react";

import {
  getContactDetailsV2,
  getProcessV1,
  mockContactDetailsV2,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, within } from "@testing-library/react";

import { locale } from "../../../services";
import {
  mockBreachChecksFailedState,
  mockBreachChecksPassedState,
  mockDocumentsRequestedDes,
  mockManualChecksPassedState,
  mockProcessAutomatedChecksFailed,
  mockProcessAutomatedChecksPassed,
  mockProcessInvalidState,
  mockProcessSelectTenants,
} from "../../../test-utils";
import { ProcessLayout } from "../process-layout";

import * as tenureV1 from "@mtfh/common/lib/api/tenure/v1/service";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";
import * as errorMessages from "@mtfh/common/lib/hooks/use-error-codes";
import { AxiosSWRResponse } from "@mtfh/common/lib/http";

jest.mock("react", () => ({
  ...jest.requireActual<any>("react"),
  useState: jest.fn(),
}));

const useStateMock: jest.Mock<typeof useState> = useState as never;

const options = {
  url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

test("it renders soletojoint view for SelectTenants", async () => {
  server.use(getProcessV1(mockProcessSelectTenants));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);

  await expect(
    screen.findByTestId("soletojoint-SelectTenants"),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps[0].className).toContain("active");
  expect(steps[1].className).not.toContain("active");
});

test("it renders stepper component", async () => {
  server.use(getProcessV1(mockProcessSelectTenants));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps).toMatchSnapshot();
});

test("it renders sidebar buttons correctly", async () => {
  server.use(getProcessV1(mockProcessSelectTenants));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);

  const { soleToJoint } = locale.views;
  const { cancelProcess, ...actions } = soleToJoint.actions;
  await Promise.all(
    Object.keys(actions).map(async (key) => {
      const button = await screen.findByText(soleToJoint.actions[key]);
      expect(button.className).toContain("secondary");
    }),
  );
  await expect(screen.queryByText(cancelProcess)).toBeNull();
});

test("it renders soletojoint view for AutomatedChecksFailed", async () => {
  server.use(getProcessV1(mockProcessAutomatedChecksFailed));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);
  await expect(
    screen.findByTestId("soletojoint-CheckEligibility"),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText(locale.views.closeProcess.outcomeLetterSent),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps[0].className).not.toContain("active");
  expect(steps[1].className).toContain("active");
});

test("it renders soletojoint view for AutomatedChecksPassed", async () => {
  server.use(getProcessV1(mockProcessAutomatedChecksPassed));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);
  await expect(
    screen.findByTestId("soletojoint-CheckEligibility"),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps[0].className).not.toContain("active");
  expect(steps[1].className).toContain("active");
});

test("it renders soletojoint view for state=ManualChecksPassed, furtherEligibilitySubmitted=false", async () => {
  server.use(getProcessV1(mockManualChecksPassedState));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);
  await expect(
    screen.findByTestId("soletojoint-ManualChecksPassed"),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps[0].className).toContain("active");
  expect(steps[0].textContent).toContain("Breach of tenure check");
});

test("it renders soletojoint view for state=ManualChecksPassed, furtherEligibilitySubmitted=true", async () => {
  const tenure = {
    householdMembers: [
      {
        fullName: "Test0 Test",
        isResponsible: false,
      },
      {
        fullName: "Test1 Test",
        isResponsible: true,
      },
      {
        fullName: "Test2 Test",
        isResponsible: true,
      },
    ],
    tenuredAsset: {
      propertyReference: "004567124",
    },
  } as Tenure;

  server.use(getProcessV1(mockManualChecksPassedState));

  jest.spyOn(tenureV1, "useTenure").mockReturnValue({
    data: tenure,
  } as AxiosSWRResponse<Tenure>);

  useStateMock
    // @ts-ignore
    .mockReturnValue([true, jest.fn()])
    // @ts-ignore
    .mockReturnValueOnce([false, jest.fn()])
    // @ts-ignore
    .mockReturnValueOnce([false, jest.fn()])
    // @ts-ignore
    .mockReturnValueOnce([false, jest.fn()])
    // @ts-ignore
    .mockReturnValueOnce([false, jest.fn()])
    // @ts-ignore
    .mockReturnValueOnce([false, jest.fn()]);
  render(<ProcessLayout />, options);
  await expect(
    screen.findByTestId("soletojoint-ManualChecksPassed"),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem", { hidden: true });
  expect(steps[2].className).toContain("active");
  expect(steps[2].textContent).toContain("Finish");
});

test("it renders soletojoint view for state=BreachChecksPassed", async () => {
  server.use(getProcessV1(mockBreachChecksPassedState));
  server.use(getContactDetailsV2(mockContactDetailsV2));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
  render(<ProcessLayout />, options);
  await expect(
    screen.findByTestId("soletojoint-RequestDocuments"),
  ).resolves.toBeInTheDocument();

  const stepper = await screen.findByTestId("mtfh-stepper-sole-to-joint");
  const steps = within(stepper).getAllByRole("listitem");
  expect(steps[1].className).toContain("active");
  expect(steps[1].textContent).toContain("Request Documents");
  await expect(
    screen.findByText(locale.components.entitySummary.tenurePaymentRef, {
      exact: false,
    }),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.queryByText(locale.views.checkEligibility.autoCheckIntro),
  ).not.toBeInTheDocument();
  await expect(screen.queryByText("Eligibility checks passed")).toBeInTheDocument();
  await expect(screen.findByText("Supporting documents")).resolves.toBeInTheDocument();
});

test("it renders soletojoint for state=BreachChecksFailed", async () => {
  server.use(getProcessV1(mockBreachChecksFailedState));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);

  await expect(
    screen.findByText(locale.components.entitySummary.tenurePaymentRef, {
      exact: false,
    }),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText(locale.views.checkEligibility.autoCheckIntro),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText("Failed breach of tenure check:"),
  ).resolves.toBeInTheDocument();
});

test("it renders an error if an invalid state is returned", async () => {
  server.use(getProcessV1(mockProcessInvalidState));
  render(<ProcessLayout />, options);

  await expect(
    screen.findByText(locale.errors.unableToFindState),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText(locale.errors.unableToFindStateDescription),
  ).resolves.toBeInTheDocument();
});

test("it renders an error if tenure details can't be fetched", async () => {
  server.use(getProcessV1("error", 500));
  render(<ProcessLayout />, options);

  await expect(
    screen.findByText(locale.errors.unableToFetchRecord),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText(locale.errors.unableToFetchRecordDescription),
  ).resolves.toBeInTheDocument();
});

test("it renders soletojoint for state=DocumentsRequestedDes", async () => {
  server.use(getProcessV1(mockDocumentsRequestedDes));
  // @ts-ignore
  useStateMock.mockImplementation(() => [false, jest.fn()]);
  render(<ProcessLayout />, options);
  await expect(
    screen.findByText(locale.components.entitySummary.tenurePaymentRef, {
      exact: false,
    }),
  ).resolves.toBeInTheDocument();
  await expect(
    screen.findByText(locale.views.reviewDocuments.passedChecks, {
      exact: true,
    }),
  ).resolves.toBeInTheDocument();
});
