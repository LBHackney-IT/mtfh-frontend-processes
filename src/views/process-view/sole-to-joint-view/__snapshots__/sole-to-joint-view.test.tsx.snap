// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders stepper component 1`] = `
Array [
  <li
    class="mtfh-stepper__step mtfh-stepper__step--active"
  >
    <div
      class="mtfh-stepper__header"
    >
      <h4
        class="lbh-heading-h4 mtfh-stepper__title"
      >
        <span
          class="mtfh-stepper__circle mtfh-stepper__circle--number"
        >
          <span
            class="mtfh-stepper__circle-inner"
          >
            <span
              class="mtfh-stepper__circle-background"
            >
              <span
                class="govuk-visually-hidden"
              >
                Step
              </span>
              1
            </span>
          </span>
        </span>
        <span>
          <button
            class="mtfh-stepper__button mtfh-stepper__button--title"
          >
            <span>
              Select person
            </span>
          </button>
        </span>
      </h4>
    </div>
  </li>,
  <li
    class="mtfh-stepper__step"
  >
    <div
      class="mtfh-stepper__header"
    >
      <h4
        class="lbh-heading-h4 mtfh-stepper__title"
      >
        <span
          class="mtfh-stepper__circle mtfh-stepper__circle--number"
        >
          <span
            class="mtfh-stepper__circle-inner"
          >
            <span
              class="mtfh-stepper__circle-background"
            >
              <span
                class="govuk-visually-hidden"
              >
                Step
              </span>
              2
            </span>
          </span>
        </span>
        <span>
          <button
            class="mtfh-stepper__button mtfh-stepper__button--title"
          >
            <span>
              Eligibility Checks
            </span>
          </button>
        </span>
      </h4>
    </div>
  </li>,
  <li
    class="mtfh-stepper__step"
  >
    <div
      class="mtfh-stepper__header"
    >
      <h4
        class="lbh-heading-h4 mtfh-stepper__title"
      >
        <span
          class="mtfh-stepper__circle mtfh-stepper__circle--number"
        >
          <span
            class="mtfh-stepper__circle-inner"
          >
            <span
              class="mtfh-stepper__circle-background"
            >
              <span
                class="govuk-visually-hidden"
              >
                Step
              </span>
              3
            </span>
          </span>
        </span>
        <span>
          <button
            class="mtfh-stepper__button mtfh-stepper__button--title"
          >
            <span>
              Finish
            </span>
          </button>
        </span>
      </h4>
    </div>
  </li>,
]
`;
